### PR TITLE
Watch and reload TLS certificates for metrics

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:
We now add new file watchers for the metrics TLS cert and key as well as
reloading them on any event. If the reload fails, then we keep the old
certificates as fallback.
#### Which issue(s) this PR fixes:

None
#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
Automatically reload metrics TLS certificate and key if any of those specified files change.
```

### CODE REVIEW

File: server/server.go [hunk 1]
```diff
@@ -644,25 +631,37 @@ func (s *Server) startMetricsServer() error {
...
+			kpr, reloadErr := newCertReloader(
+				s.monitorsChan, s.config.MetricsCert, s.config.MetricsKey,
+			)
+			if reloadErr != nil {
+				logrus.Fatalf("Creating key pair reloader: %v", reloadErr)
+			}
+
+			srv := http.Server{
+				Handler:   me,
+				TLSConfig: &tls.Config{GetCertificate: kpr.getCertificate},
```
[the remark this answers is not in the document]
Comment (saschagrunert) [reply server/server.go:662]: Fixed

File: server/server.go [hunk 2]
```diff
@@ -676,6 +678,83 @@ func startMetricsEndpoint(network, address, cert, key string, me *http.ServeMux)
 	return nil
 }
 
+type certReloader struct {
```
Comment (haircommander) [inline server/server.go:681]: what would you think about moving this to a separate file? server/server.go is pretty big already

Comment (haircommander) [reply server/server.go:681]: maybe server/metrics.go?

Comment (saschagrunert) [reply server/server.go:681]: I'd like to move the whole metrics part into another file/struct. I can follow up on this if you want. Or should I do it within this PR?

Comment (saschagrunert) [reply server/server.go]: Sounds good, moved everything over there.

**Comment by haircommander** [comment]
/retest
/approve

LGTM

**Comment by saschagrunert** [comment]
/retest
@haircommander PTAL

**Pull request merged**